build.sbt

```diff
@@ -196,6 +198,26 @@ lazy val zioJsonYaml = project
   )
   .dependsOn(zioJsonJVM)
 
+lazy val zioJsonMacros = crossProject(JSPlatform, JVMPlatform)
+  .in(file("zio-json-macros"))
+  .settings(stdSettings("zio-json-macros"))
+  .settings(crossProjectSettings)
+  .enablePlugins(NeoJmhPlugin)
+  .settings(macroExpansionSettings)
+  .settings(
+    scalacOptions -= "-Xfatal-warnings", // not quite ready.
+    libraryDependencies ++= Seq(
+      "org.scala-lang" % "scala-reflect" % scalaVersion.value % Provided,
+      "dev.zio"      %%% "zio-test"      % zioVersion         % "test",
+      "dev.zio"      %%% "zio-test-sbt"  % zioVersion         % "test"
+    ),
+    testFrameworks += new TestFramework("zio.test.sbt.ZTestFramework")
+  )
+
+lazy val zioJsonMacrosJVM = zioJsonMacros.jvm.dependsOn(zioJsonJVM)
```

Comment:
These extracted vals are no longer necessary.
The `.dependsOn` sections can also be defined together with the project using `.jsConfigure` / `.jvmConfigure`

Reply:
@fsvehla the extracted vals are required because they need to be referred in root project to be able to publish them on maven.